### PR TITLE
Improving the dropdown dialog widget documentation 

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this PR

This PR improves the dropdown widget documentation with some useful options which can be used on widget's initialization, also a good example which can be used for widget's initialization was provided.

## Affected URLs

- https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/widgets/widget_dialog.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget_dialog.html
- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_dialog.html

## Links to source code
- Codebase - https://github.com/magento/magento2/blob/2.0/lib/web/mage/dropdown.js

whatsnew
Added a [new code example](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_dialog.html) showing how to initialize and pass options to the dropdown dialog widget.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/javascript-dev-guide/widgets/widget_dialog.md [hunk 1]
```diff
@@ -146,6 +157,43 @@ Beside default functionality, this method calls the `_mouseLeave()` and `_mouseE
...
+        Here is our content
+    </div>
+</div>
+```
+
+### Result
+
+The result is a dropdown dialog and a button (_Click Here_) that opens the dialog.
+
+![DropdownDialog Widget]({{ site.baseurl }}/common/images/widget/dropdown-dialog-widget-result.png)
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_dialog.md]: According to this PR, you actually added the image to the following location:

```guides/v2.1/javascript-dev-guide/widgets/images/dropdown-dialog-widget-result.png```

Please move the image to:

```common/images/widget/dropdown-dialog-widget-result.png```

**Comment by eduard13** [comment]
Hi @jeff-matthews, you're right. I've jut moved it into the right folder. 
Thank you.

**Comment by dobooth** [comment]
running tests

**Pull request merged**